src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+  const editAvatarCloseButton = document.querySelector('#edit-avatar-close-btn');
+  const editAvatarForm = editAvatarModal.querySelector('form');
+  const editAvatarLinkInput = editAvatarModal.querySelector('#edit-avatar-link-input');
+  const avatarSubmitBtn = editAvatarModal.querySelector('.modal__submit-btn');
+  const profileAvatar = document.getElementById('profile-avatar');
+
+  function closeAvatarModal() {
+    closeModal(editAvatarModal);
+  }
+
+  const openAvatarModal = () => {
+    resetValidation(editAvatarForm, [editAvatarLinkInput]);
+    openModal(editAvatarModal);
+  };
+
+
+  function handleEditAvatarSubmit(evt) {
+    evt.preventDefault();
+
+    if (!editAvatarForm.checkValidity()) {
+      return;
+    }
+
+    const avatarUrl = editAvatarLinkInput.value;
+
+    avatarSubmitBtn.textContent = 'Saving...';
+    avatarSubmitBtn.disabled = true;
+
+    api.editUserAvatar(avatarUrl)
+      .then((userData) => {
+        // Ensure that userData includes the correct avatar URL
+        profileAvatar.src = userData.avatar;
+
+        closeAvatarModal();
+        avatarSubmitBtn.textContent = 'Save';
+        avatarSubmitBtn.disabled = false;
+      })
+      .catch((err) => {
+        console.error('Error updating avatar:', err);
+
+        avatarSubmitBtn.textContent = 'Save';
+        avatarSubmitBtn.disabled = false;
+      });
+  }
+
+
+
+  // Delete Modal
+  const deleteModal = document.querySelector('#delete-card-modal');
+  const deleteModalConfirmButton = deleteModal.querySelector('.modal__delete-btn');
+  const deleteModalCancelButton = deleteModal.querySelector('.modal__cancel-btn');
+  const deleteModalCloseButton = deleteModal.querySelector('.modal__close-btn_type_delete');
+
+  let cardToDelete = null;
+
+  function openDeleteModal(cardElement) {
+    cardToDelete = cardElement;
+    openModal(deleteModal);
+  }
+
+
+  function handleDeleteCard() {
+    if (cardToDelete) {
+      const cardId = cardToDelete.dataset.cardId;
+
+      api.deleteCard(cardId)
+        .then(() => {
+          cardToDelete.remove();
+          closeModal(deleteModal);
+        })
+        .catch(err => {
+          console.error('Error deleting card:', err);
+        });
+    }
+  }
+
+
+  // Cards
+  const cardTemplate = document.querySelector('#card-template').content;
+  const cardsContainer = document.querySelector('.cards__list');
+
+  api.getAppInfo()
+    .then(([userInfo, cards]) => {
+      document.querySelector('.profile__name').textContent = userInfo.name;
+      document.querySelector('.profile__description').textContent = userInfo.about;
+      document.getElementById('profile-avatar').src = userInfo.avatar;
+
+      renderInitialCards(cards);
+    })
+    .catch(err => {
+      console.error('Error receiving user information:', err);
+    });
+
+  function renderInitialCards(cards) {
+    cards.forEach((card) => {
+      const cardElement = getCardElement(card);
+      cardsContainer.appendChild(cardElement);
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend` but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardList[ method ](cardElement);
}



```